File: codegen/codegen-erlang/src/test/resources/ir/client_dispatch_list_widgets_page.expected.erl
Req = paginated_service_rest_json_1:encode_list_widgets_request(Input),
case runtime_http:dispatch(Config, Req) of
    {ok, Resp} ->
        case paginated_service_rest_json_1:decode_list_widgets_response(Resp) of
            {ok, Output} ->
                NewAcc = Acc ++ Output#list_widgets_output.widgets,
                case Output#list_widgets_output.next_token of
                    undefined ->
                        {ok, NewAcc};
                    NextToken ->
                        NextInput = Input#list_widgets_input{ next_token = NextToken },
                        list_widgets(Config, NextInput, NewAcc)
                end;
            {error, Reason} ->
                {error, Reason}
        end;
    {error, Reason} ->
        {error, Reason}
end.

File: codegen/codegen-elixir/src/test/resources/ir/client_dispatch_list_widgets_page.expected.ex
req = PaginatedServiceRestJson1.encode_list_widgets_request(input)
case RuntimeHttp.dispatch(config, req) do
  {:ok, resp} ->
    case PaginatedServiceRestJson1.decode_list_widgets_response(resp) do
      {:ok, output} ->
        new_acc = acc ++ output.widgets || []
        case output.next_token do
          nil -> {:ok, new_acc}
          next_token ->
            next_input = %{input | next_token: next_token}
            list_widgets(config, next_input, new_acc)
        end
      {:error, reason} -> {:error, reason}
    end
  {:error, reason} -> {:error, reason}
end
